libcaf_core/caf/flow/observable.test.cpp

```diff
@@ -68,6 +68,106 @@ TEST("element_at(n) only takes the element with index n") {
   }
 }
 
+TEST("reduce(z, f) applies a function to each item, sequentially, and emits "
+     "the final value") {
```

Comment:
```suggestion
TEST("reduce(init, fn) combines all items into one final value using fn") {
  std::plus<int> adder;
```

(and then pass `adder` everywhere below instead of constructing a `std::plus` object each time.